src/components/ToDoContainer/index.js

```diff
@@ -1,9 +1,103 @@
 import React, { Component } from "react";
+import AddTodo from "./AddTodo/index.js";
+import TodoItems from "./TodoItems/index.js";
+import SearchTodos from "./SearchTodos/index.js";
+import { Container, Text, SubContainer } from "./styles";
 
 class ToDoContainer extends Component {
+  // Set up the state
+  constructor() {
+    super();
+    this.state = {
+      items: [], // Will store all the todos and display
+      currentText: "", // Todo inputted by user
+      currentID: 1,
+      searchText: ""
+    };
+  }
+
+  // Update what the current item is based on user input
+  handleInput = item => {
+    const todo = item.target.value;
+    const currentText = todo;
+    this.setState({ currentText });
+  };
+
+  // Add an item to the list
+  addTodo = obj => {
+    obj.preventDefault();
+
+    // Make sure the current item is not empty before adding
+    if (this.state.currentText !== "") {
+      const items = [
+        ...this.state.items,
+        {
+          id: this.state.currentID + 1,
+          text: this.state.currentText,
+          status: true
+        }
+      ];
+      this.setState({
+        items: items,
+        currentText: "", // Reset text the next user input
+        currentID: this.state.currentID + 1 // New ID value
+      });
+    }
+  };
+
+  // Delete an item from the list
+  deleteTodo = id => {
+    // id.preventDefault();
+
+    // Filter to create new array that doesn't include deleted item
+    const newList = this.state.items.filter(obj => {
+      return obj.id !== id;
+    });
+
+    this.setState({
+      items: newList
+    });
+  };
+
+  searchTodos = input => {
+    const matches = this.state.items.filter(t => {
+      t.text.match(input) !== null ? (t.status = true) : (t.status = false);
+      return t;
+    });
+
+    this.setState({
```

Comment:
By setting the state of the items to only the matching todos of the query, we lose all of the other todos even after the user gets rid of their text. Instead of changing the state when processing the queries, I'd think about a way to filter which todos get displayed to the user more dynamically